test/e2e/service_externalip_test.go

```diff
@@ -368,6 +369,68 @@ func testServiceWithExternalIPCRUD(t *testing.T, data *TestData) {
 	}
 }
 
+func testServiceSharingLoadBalancerIP(t *testing.T, data *TestData) {
+	pool := data.createExternalIPPool(t, "pool-", v1beta1.IPRange{CIDR: "172.30.0.0/28"}, nil, nil, nil)
+	defer data.crdClient.CrdV1beta1().ExternalIPPools().Delete(context.TODO(), pool.Name, metav1.DeleteOptions{})
+
+	annotationNormal := map[string]string{
+		antreaagenttypes.ServiceExternalIPPoolAnnotationKey: pool.Name,
+	}
+	annotationAllowingSharedIP := map[string]string{
+		antreaagenttypes.ServiceExternalIPPoolAnnotationKey: pool.Name,
+		antreaagenttypes.ServiceAllowSharedIPAnnotationKey:  "true",
+	}
+	loadBalancerIPMutator := func(svc *v1.Service) { svc.Spec.LoadBalancerIP = "172.30.0.1" }
+
+	t.Run("services-allowing-shared-ip", func(t *testing.T) {
+		svc1, err := data.CreateServiceWithAnnotations("svc1",
+			data.testNamespace, 80, 80, v1.ProtocolTCP, nil, false, false, v1.ServiceTypeLoadBalancer, nil, annotationAllowingSharedIP,
+			loadBalancerIPMutator)
+		require.NoError(t, err)
+		defer data.clientset.CoreV1().Services(svc1.Namespace).Delete(context.TODO(), svc1.Name, metav1.DeleteOptions{})
+
+		require.NoError(t, data.waitForServiceLoadBalancerIP(svc1, "172.30.0.1"))
+
+		svc2, err := data.CreateServiceWithAnnotations("svc2",
+			data.testNamespace, 81, 80, v1.ProtocolTCP, nil, false, false, v1.ServiceTypeLoadBalancer, nil, annotationAllowingSharedIP,
+			loadBalancerIPMutator)
+		require.NoError(t, err)
+		defer data.clientset.CoreV1().Services(svc2.Namespace).Delete(context.TODO(), svc2.Name, metav1.DeleteOptions{})
+		svc3, err := data.CreateServiceWithAnnotations("svc3",
+			data.testNamespace, 82, 80, v1.ProtocolTCP, nil, false, false, v1.ServiceTypeLoadBalancer, nil, annotationNormal,
+			loadBalancerIPMutator)
+		require.NoError(t, err)
+		defer data.clientset.CoreV1().Services(svc3.Namespace).Delete(context.TODO(), svc3.Name, metav1.DeleteOptions{})
+
+		require.NoError(t, data.waitForServiceLoadBalancerIP(svc2, "172.30.0.1"), "svc2 should get the shared IP assigned")
+		require.NoError(t, data.waitForServiceLoadBalancerIP(svc3, ""), "svc3 should not get the shared IP assigned")
```

Comment:
nit: I would use `assert` here since we could check the second assertion even if the first one fails and that could provide extra information in case of failure?

Reply:
makes sense, done